core/src/main/java/org/frankframework/pipes/IsUserInRolePipe.java

```diff
@@ -63,28 +65,38 @@ public void configure() throws ConfigurationException {
 		}
 	}
 
-	protected void assertUserIsInRole(PipeLineSession session, String role) throws SecurityException {
-		if (!session.isUserInRole(role)) {
-			throw new SecurityException("user is not in role ["+role+"]");
+	protected String inWhichRoleIsUser(PipeLineSession session, String roles) {
+		List<String> splitRoles = StringUtil.split(roles);
+		return session.inWhichRoleIsUser(splitRoles);
+	}
+
+	protected String getRolesToCheck(Message message) throws PipeRunException {
+		if (StringUtils.isEmpty(getRole())) {
+			try {
+				return message.asString();
+			} catch (IOException e) {
+				throw new PipeRunException(this, "cannot open stream", e);
+			}
 		}
+		return getRole();
 	}
 
 	@Override
 	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
 		try {
-			if (StringUtils.isEmpty(getRole())) {
-				String inputString;
-				try {
-					inputString = message.asString();
-				} catch (IOException e) {
-					throw new PipeRunException(this, "cannot open stream", e);
-				}
-				if (StringUtils.isEmpty(inputString)) {
-					throw new PipeRunException(this, "role cannot be empty");
-				}
-				assertUserIsInRole(session, inputString);
-			} else {
-				assertUserIsInRole(session, getRole());
+			String rolesToCheck = getRolesToCheck(message);
+			if (StringUtils.isEmpty(rolesToCheck)) {
+				throw new PipeRunException(this, "role cannot be empty");
+			}
+
+			String userRole = inWhichRoleIsUser(session, rolesToCheck);
+			if (userRole == null) {
+				throw new SecurityException("user is not in role(s) [" + rolesToCheck + "]");
+			}
+
+			PipeForward relatedForward = findForward(userRole);
```

Comment:
In the case where the user has multiple roles, the first specified and matching role is taken.

Reply:
That is very pipe specific but I don't see a better way to do it.
The results might be unpredictable though.

Reply:
What makes the results very unpredictable if this behaviour is properly documented? The exact same message will always produce the exact same result.

Reply:
Only if you can guarantee that all the security handler implementations will always enumerate the roles in the same order.
Which you cannot in the long run because that is not contractually guaranteed.

So if you want predictability, best to fetch all matching roles in this pipe, sort the user-roles alphabetically by role-name, then pick the first one.

Reply:
We iterate over the roles to check, not the roles present, the order of the roles to check is always respected.

By alphabetically sorting the role names, the Frank developer has lost the ability to prioritize certain roles over others, by specifying them first...

Reply:
OK, good point. So no need to sort alphabetically as long as the pipe checks roles in order.

Another argument for me though to put this in the pipe and better document + test this behaviour. For generic code, I might otherwise be very tempted at some point in future to replace the check-code with a simple set-intersection. And then you lose all ordering guarantees.